Adafruit_MQTT.cpp

```diff
@@ -508,10 +508,32 @@ void Adafruit_MQTT::processPackets(int16_t timeout) {
   }
 }
 Adafruit_MQTT_Subscribe *Adafruit_MQTT::readSubscription(int16_t timeout) {
-  // Check if data is available to read.
-  uint16_t len =
-      readFullPacket(buffer, MAXBUFFERSIZE, timeout); // return one full packet
-  return handleSubscriptionPacket(len);
+
+  // Sync or Async subscriber with message
+  Adafruit_MQTT_Subscribe *s = 0;
+
+  // Check if are unread messages
```

Comment:
Isn't this handled within `handleSubscriptionPacket`? Why is the functionality duplicated here?

Reply:
handleSubscriptionPacket reads the incomming message but readSubscription expects for a new message, losing the packet that come in processPacketsUntil while waiting for a ping or publish response

Reply:
Ok! Off-topic but is there a way to recover this lost packet that you know of?